File: frontend/src/lib/api/ingestion.ts
/*
 1. FILE PURPOSE: API calls for ingestion (upload scripts + answer key).
 2. RESPONSIBILITIES:
    - uploadScripts(batchId, files); uploadAnswerKey(batchId, keyFile).
 3. PLANNED CONTENTS: async functions using the shared axios client.
 4. INPUTS / OUTPUTS: Inputs: files/ids. Outputs: typed ingest responses.
 5. DEPENDS ON / USED BY: lib/api/client, types; used by hooks/useEvaluation / ingestion page.
*/

File: backend/app/api/routes/ingestion.py
# 1. FILE PURPOSE: API routes for ingesting scanned answer scripts and reference material.
# 2. RESPONSIBILITIES:
#    - POST /ingest — accept uploaded PDFs/images for a batch and start preprocessing.
#    - POST /answer-key — upload the question paper + answer key + rubric for a batch.
# 3. PLANNED CONTENTS: FastAPI endpoint functions delegating to services/evaluation_service + app/ingestion/*.
# 4. INPUTS / OUTPUTS: Inputs: multipart uploads, batch id. Outputs: JSON ingest status / stored key metadata.
# 5. DEPENDS ON / USED BY: fastapi, api/schemas.py, services/evaluation_service.py, app/ingestion/*.
